### PR TITLE
Add support ESLint 8

### PR DESCRIPTION
**Pull request by sndrs**
## What does this change?

- upgrades to ESLint 8
- upgrades to prettier 2.4 
  - `jsxBracketSameLine` is deprecated
- makes `eslint` an explicit peerDep of `@guardian/eslint-config-typescript`
- pins `@guardian/eslint-config-typescript`'s dependency on `@guardian/eslint-config` to a specific version
- peerdeps more accurately express the widest possible range
- installs the minimum version of peerdeps we support as dev deps
- removed unnecessary devDeps

also:

- use yarn workspaces (rather than lerna's)
	- this give us better feedback from yarn about peerDep irregularities

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,15 +12,9 @@
...
-    "eslint": "^7.28.0",
-    "eslint-config-prettier": "^8.3.0",
-    "eslint-plugin-eslint-comments": "^3.2.0",
-    "eslint-plugin-import": "^2.23.4",
-    "eslint-plugin-prettier": "^3.4.0",
+    "eslint": "8.0.0",
     "lerna": "^4.0.0",
-    "prettier": "^2.3.1",
-    "typescript": "^4.3.2"
+    "prettier": "2.4.0",
```
Comment (SiAdcock) [inline package.json:17]: Nit: is there a reason we must pin to this specific version of `prettier`, or can we specify a `^` range in keeping with  other devDeps?

File: packages/prettier/package.json [hunk 2]
```diff
@@ -13,7 +13,7 @@
     "url": "https://github.com/guardian/configs/issues"
   },
   "peerDependencies": {
-    "prettier": "^2.0.0"
+    "prettier": "^2.4.0"
```
Comment (SiAdcock) [inline packages/prettier/package.json:16]: Why force consumers to upgrade? Are we using features that were added after 2.0.0?

File: package.json [hunk 3]
```diff
@@ -9,15 +12,9 @@
...
     "@guardian/eslint-config-typescript": "file:packages/eslint-config-typescript",
     "@guardian/prettier": "file:packages/prettier",
-    "@typescript-eslint/eslint-plugin": "^4.26.0",
-    "@typescript-eslint/parser": "^4.26.0",
-    "eslint": "^7.28.0",
-    "eslint-config-prettier": "^8.3.0",
-    "eslint-plugin-eslint-comments": "^3.2.0",
-    "eslint-plugin-import": "^2.23.4",
-    "eslint-plugin-prettier": "^3.4.0",
+    "eslint": "8.0.0",
```
Comment (akash1810) [inline package.json:15]: [8.4.1](https://www.npmjs.com/package/eslint) is the latest, should we be using that instead?

File: package.json (hunk 1 above)
Comment (sndrs) [reply package.json:17]: 2.4 has the new brackets flag config, so this becomes the minimum version the workspace should work with

File: packages/prettier/package.json (hunk 2 above)
Comment (sndrs) [reply packages/prettier/package.json:16]: Yeah the new new brackets config comes with 2.4

File: package.json (hunk 3 above)
Comment (sndrs) [reply package.json:15]: The idea was to install the minimum possible version in the workspace that our packages should work with, to make using them as flexible as possible without accidentally using features added after the minimum version

File: package.json (hunk 1 above)
Comment (SiAdcock) [reply package.json:17]: I get ya. So the general rule is:

- in peerDependencies, specify the minimum supported version of a package, using a `^` to pin the major version and enforce a minimum minor version
- in devDependencies,  specify the minimum supported version of a package, using the exact version (i.e. no `^` or `~`)

**Review by SiAdcock — APPROVED** [review]
This is cool. I think it'll be a breaking change, so major bump when we publish?

File: package.json (hunk 1 above)
Comment (sndrs) [reply package.json:17]: that's was the thinking yeah – make it as flexible as possible to consume, but make sure we develop with the smallest possible feature set we need. and then obvs if we need a feature from a later version that becomes the baseline

**Pull request merged**